homeassistant/components/generic_water_heater/__init__.py

```diff
@@ -0,0 +1,54 @@
+"""The generic_water_heater integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.water_heater import DOMAIN as WATER_HEATER_DOMAIN
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "generic_water_heater"
+
+CONF_HEATER = "heater_switch"
+CONF_SENSOR = "temperature_sensor"
+CONF_TARGET_TEMP = "target_temperature"
+CONF_TEMP_DELTA = "delta_temperature"
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                cv.slug: vol.Schema(
+                    {
+                        vol.Required(CONF_HEATER): cv.entity_id,
+                        vol.Required(CONF_SENSOR): cv.entity_id,
+                        vol.Optional(CONF_TEMP_DELTA): vol.Coerce(float),
+                        vol.Optional(CONF_TARGET_TEMP): vol.Coerce(float),
+                    }
+                )
+            }
```

Comment:
Can we flatten the schema so its more like https://www.home-assistant.io/integrations/generic_thermostat/

Reply:
not sure I understand the comment, but:

`generic_thermostat` uses old style configuration, hanging itself from `climate:` and then `platform: generic_thermostat`

as for `generic_water_heater` its configuration hangs from the top level of configuration `generic_water_heater:`

therefore the difference between the two.

Reply:
I'm suggesting flatting the schema to be a `list` of `dict`s instead of a `dict` of `dict`s:

```
generic_water_heater:
 - name: bath water
   heater_switch: switch.dhw_switch
   temperature_sensor: sensor.dhw_temperature
   target_temperature: 50
   delta_temperature: 5
 - name: master water
   heater_switch: switch.master_switch
   temperature_sensor: sensor.master_temperature
   target_temperature: 50
   delta_temperature: 5
```

Reply:
I understand your comment now,

I believe to be a question of preferred style, my approach was similar to `alexa`, `utility_meter`, `counter` and I believe others (these examples come from my configuration.yaml 😄)

Reply:
I usually try to avoid unvalidated user input in hash keys since it's been a source of pain in the past (especially when users put non printable characters in the key)

This isn't a showstopper, just a suggestion